task_processing/interfaces/event.py

```diff
@@ -8,18 +8,36 @@
 
 
 class Event(PRecord):
-    timestamp = field(type=datetime.datetime)
+    type = field(type=str, initial=None)
```

Comment:
also none of this is set in stone, point of this PR is to have a discussion and how to represent event hierarchy is very valid question. let's hear more opinions on this?